src/backend/external_loop_closure_receiver.cpp

```diff
@@ -134,6 +136,55 @@ LookupResult ExternalLoopClosureReceiver::findClosest(const DynamicSceneGraph& g
   return {LookupResult::Status::VALID, best_id};
 }
 
+bool should_add_lc(const std::set<pose_graph_tools::PoseGraphEdge,
+                                  decltype(pose_graph_edge_cmp)*>& added_lcs,
+                   const uint64_t stamp_ns_from,
+                   const uint64_t stamp_ns_to,
+                   const Eigen::Affine3d& to_T_from,
+                   const uint64_t lockout_time_ns,
+                   const double min_pose_discrepancy) {
+  auto pge_stamp_cmp = [](const pose_graph_tools::PoseGraphEdge& pge,
+                          const uint64_t value) { return pge.key_from < value; };
+
+  auto lc_iter = std::lower_bound(added_lcs.begin(),
+                                  added_lcs.end(),
+                                  stamp_ns_from - lockout_time_ns,
```

Comment:
Potential overflow issue if `lockout_time_ns` is really big (probably should be using signed representations for this)